dart/lib/src/protocol/sentry_exception.dart

```diff
@@ -29,7 +29,7 @@ class SentryException {
     this.module,
     this.stackTrace,
     this.mechanism,
-    this.threadId,
+    this.threadId = 0,
```

Comment:
only if its a mandatory field, then 0, because isolates don't have an id right

Reply:
But then how do we link threads to exceptions?

Reply:
there's always only 1 thread, right? so we can default to 0?

Reply:
Well, you can have different isolates, but you can't identify them with an id of some sorts. Also, because the user is responsible for collecting exceptions in another isolate, I would assume that it's safe to default to 0.

Reply:
@ueman we'd only have a single isolate in the `threads` list or many? Could we differentiate between the currently running one where the exception was created? If so we should use their `id` and if no `id` exist we can do like Manoel suggested and add an artificial `id` to the exception and the the 1 isolate that did have an issue.

We just need to keep in mind if we're merging data (adding Java threads for example) things can get messy but AFAIK we don't merge that

Reply:
Shouldn't this default to `null`?

Are we going to always send a threads list by default so the concern is that we'll add the thread's stack-trace instead of the exception? If that's the case we need to match the id so Sentry's UI prioritizes the exceptions'.

But if we're adding threads but not sending it (then are we adding this for some cross-SDK support feature?) it would be be better to ensure this is `null`

Reply:
> we'd only have a single isolate in the threads list or many?

We can't automatically populate `threads` with other isolates except for the current isolate. The user could do that though.

> Could we differentiate between the currently running one where the exception was created?

No, isolates don't have ids and exceptions don't propagate from one isolate to another.

> Are we going to always send a threads list by default so the concern is that we'll add the thread's stack-trace instead of the exception?

Kinda, what I want to do is to only send a stacktrace with the exception if `options.attachStacktrace` is true, so that we don't add the same stacktrace two times.

Reply:
> Kinda, what I want to do is to only send a stacktrace with the exception if options.attachStacktrace is true, so that we don't add the same stacktrace two times.

If `attachStacktrace` is true, we create a thread with stacktrace, not an exception though.
If we're capturing an exception (with a stacktrace) then the `attachStacktrace` option doesn't play a role. It's useful for a `captureMessage` for example.


> No, isolates don't have ids and exceptions don't propagate from one isolate to another.

I'm just trying to push for not defauling to `0`. It's best not to send anything then sending `0` on each event. If we don't know the `id` lets not send any?